### PR TITLE
MRG+2: Revise install docs

### PR DESCRIPTION
**Pull request by drammock**
This is the first in a series of PRs to improve the documentation.  This PR targets the installation/setup docs.  Highlights:

- converts the Install landing page to a full TOC of all the install pages

- adds a "before you install" page to guide new users about which tools they need and an overview of what each tool does.

- adds a dedicated page on freesurfer installation.

- adds "next" links to the bottom of the main installation pages.  (for now I've avoided trying to hack the sphinx-bootstrap theme to generate these automatically, like the [sphinx-readthedocs theme](https://sphinx-rtd-theme.readthedocs.io/en/latest/) does, but I'm open to that possibility if there's general support for it.)

- attempts to avoid "MNE" in favor of explicit reference to "MNE-C", "MNE-python", or "the MNE MATLAB toolbox".  When reference to all tools in general is needed, I use "the MNE suite of software tools" or similar.

- re-orders `advanced_setup` page topics to follow what I'm guessing is most -> least frequently used topics (i.e., puts ipython/jupyter first).

- describes miniconda alongside anaconda as a valid setup option.

- overhauls the MNE-C install instructions (tightens up some parts, omits others, adds info about missing libxp6 and gfortran).

- a few of minor changes outside the "installation/setup" docs that I noticed along the way:
    - `index.rst`: closing tag now matches opening tag
    - `documentation.rst`: reflow the source text
    - `contributing.rst`: make title more consistent with install pages.

cc #5883 (although this PR doesn't overlap with #5883 much, subsequent ones likely will).

### CODE REVIEW

**Comment by drammock** [comment]
artifacts: https://11145-1301584-gh.circle-artifacts.com/0/dev/getting_started.html

**Comment by drammock** [comment]
artifacts: https://11150-1301584-gh.circle-artifacts.com/0/dev/getting_started.html

File: doc/getting_started.rst [hunk 1]
```diff
@@ -2,76 +2,18 @@
...
-      However, MNE can read native CTF, BTI/4D, KIT and various
-      EEG formats (see :ref:`IO functions <ch_convert>`).
-
-      If you have been using MNE-C, there is no need to convert your fif
-      files to a new system or database -- MNE works nicely with
-      the historical fif files.
+    pre_install
+    install_freesurfer
+    install_mne_c
+    install_mne_python
```
Comment (larsoner) [inline doc/getting_started.rst]: I would change the order here, `install_mne_python` then `install_freesurfer` then `install_mne_c`

File: doc/install_mne_python.rst [hunk 2]
```diff
@@ -2,101 +2,112 @@
...
+MNE-python 0.18 only supports Python version 3.5 or higher. We strongly
+recommend the `Anaconda`_ or `Miniconda`_ distributions of python. The main
+difference is that Anaconda comes with more than 250 scientific packages
+pre-bundled, whereas Miniconda starts off with a minimal set of around 30
+packages. Both distributions include the ``conda`` command line tool for
+installing new packages and managing different package sets ("environments")
+for different projects. Follow the installation instructions for `Anaconda
+<anaconda-install>`__ or `Miniconda <miniconda-install>`__; when you are done,
+you should see a similar output if you type the following command in a
+terminal:
```
Comment (larsoner) [inline doc/install_mne_python.rst]: I wouldn't add stuff about `miniconda` here. There are a ton of ways to install Python in general, and MNE works with most of them. I'd rather promote one path here, which is what 95% of users will want. The 5% are advanced enough to learn about other methods (miniconda included) themselves.

If you really want to keep info about it, let's add something to the advanced install/debugging page about how you could use miniconda. Eventually we can mentin that `pip install --user requirements.txt` with Linux generally works, too, if we want, but stuff like that can be separate.

**Comment by larsoner** [comment]
@choldgraf you might be interested in the restructuring elements here

File: doc/install_mne_python.rst [hunk 3]
```diff
@@ -2,101 +2,112 @@
...
 
-  If you are on macOS, you need to manually update PyQt5. This step is not needed on Linux, and even breaks things on Windows.
+  If you are on macOS, you need to manually update PyQt5. This step is not
+  needed on Linux, and even breaks things on Windows.
 
   .. code-block:: console
 
     $ pip install --upgrade "pyqt5>=5.10"
 
+Testing your installation
```
Comment (drammock) [inline doc/install_mne_python.rst]: I've streamlined this section a little bit. Instead of

```
$ python
>>> import mne
```

I've converted the initial install test to `python -c 'import mne; mne.sys_info()'` and merged the subsequent discussion about `sys_info()` here

File: doc/pre_install.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
+.. include:: links.inc
```
Comment (drammock) [inline doc/pre_install.rst]: this whole file is new.  Some of the info is from the "historical note" on the old installation landing page. Some I added (like links to MNE-HCP and MNE-BIDS).  The section about "what should I install?" is probably the part where other sets of eyes are most needed.

File: doc/install_mne_c.rst [hunk 5]
```diff
@@ -2,290 +2,162 @@
 
 .. _install_mne_c:
 
-Install MNE-C
--------------
+Installing MNE-C
```
Comment (drammock) [inline doc/install_mne_c.rst]: this document changed a lot.  I went with the "make bold edits" approach in an effort to make the page substantially shorter.  Happy to discuss further / put some things back, if there's opposition.

File: doc/install_mne_c.rst [hunk 6]
```diff
@@ -2,290 +2,162 @@
...
-The MNE software is distributed as a compressed tar archive
-(Mac OSX and LINUX) or a Mac OSX disk image (dmg).
-
-The file names follow the convention:
-
-::
+System requirements
+^^^^^^^^^^^^^^^^^^^
 
-  MNE-* <*version*>*- <*rev*> -* <*Operating system*>*-* <*Processor*>*.* <*ext*>*
```
Comment (drammock) [inline doc/install_mne_c.rst]: this line (and the explanatory paragraph that followed it) are, in my opinion, a sufficiently widespread practice that there's no need to spell it out here.  Moreover, it's possible to install and use the software without ever knowing what specific version number you've installed, so there's little risk in omitting this info.

File: doc/install_mne_c.rst [hunk 7]
```diff
@@ -2,290 +2,162 @@
...
-
-- Double click the the package file. The installer starts.
-
-- Follow the instructions in the installer.
-
-.. note::
-
-    The software will be installed to /Applications/ <*name*> by default.
-    If you want another location, select Choose Folder... on the Select a
-    Destination screen in the installer.
```
Comment (drammock) [inline doc/install_mne_c.rst]: *how to click the "Choose Folder" button during macOS installation to change install location* is general software advice, not specific to MNE

File: doc/install_mne_c.rst [hunk 8]
```diff
@@ -2,290 +2,162 @@
...
-will be set in your current shell in order to run the MNE Software.
-
-A note about source-ing scripts with "source" and "." (dot). The hash bang at
-the top of a script tells exec what shell to use by default to interpret the
-script. But the hash bang line is ignored when you source a script via either
-the builtin "source" or "." directive.  So there needs to be a setup script
-compatible with your current shell, i.e., if you use csh you cannot source a
-bash script.  (BTW, the "." (dot) directive will search for the script in
-$PATH if there is no slash in the script argument while "source" does not
-search $PATH).
```
Comment (drammock) [inline doc/install_mne_c.rst]: this paragraph is general software information / advice, not specific to MNE.  If a user follows the instructions, they don't actually need to know this.

File: doc/install_mne_c.rst [hunk 9]
```diff
@@ -2,290 +2,162 @@
...
-
-
-For BEM mesh generation using the watershed algorithm or
-on the basis of multi-echo FLASH MRI data (see :ref:`create_bem_model`) and
-for accessing the tkmedit program
-from mne_analyze, see :ref:`CACCHCBF`,
-the MNE software needs access to a FreeSurfer license
-and software. Therefore, to use these features it is mandatory that
-you set up the FreeSurfer environment
-as described in the FreeSurfer documentation.
```
Comment (drammock) [inline doc/install_mne_c.rst]: this paragraph has been moved into the "system requirements" section above.

File: doc/install_mne_c.rst [hunk 10]
```diff
@@ -2,290 +2,162 @@
...
 .. code-block:: console
 
-    $ sudo add-apt-repository "deb http://security.ubuntu.com/ubuntu precise-security main"
-    $ sudo apt -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true update
-    $ sudo apt -o APT::Get::AllowUnauthenticated=true install libxp6
-    $ sudo add-apt-repository -r "deb http://security.ubuntu.com/ubuntu precise-security main"
-    $ sudo apt update
-
-.. _BABDBCJE:
+    $ sudo dpkg -i libxp6_1.0.2-1ubuntu1_amd64.deb
```
Comment (drammock) [inline doc/install_mne_c.rst]: personally I think the download-and-use-dpkg approach is preferable to asking users to change their `apt` security policies, but I could be wrong.

File: doc/install_mne_c.rst [hunk 11]
```diff
@@ -2,290 +2,162 @@
...
 
-Additional options
-##################
-
-Additional software
```
Comment (drammock) [inline doc/install_mne_c.rst]: this section merged with the "system requirements" above

File: doc/install_mne_c.rst [hunk 12]
```diff
@@ -2,290 +2,162 @@
...
-and software. Therefore, to use these features it is mandatory that
-you set up the FreeSurfer environment
-as described in the FreeSurfer documentation.
-
-The environment variables relevant to the MNE software are
-listed in :ref:`CIHDGFAA`.
-
-.. tabularcolumns:: |p{0.3\linewidth}|p{0.55\linewidth}|
-.. _CIHDGFAA:
-.. table:: Environment variables
```
Comment (drammock) [inline doc/install_mne_c.rst]: I ditched the table, since 4 of the 6 variables are discussed above, and the other two are discussed on the `mne_process_raw` page.

File: doc/install_mne_c.rst [hunk 13]
```diff
@@ -2,290 +2,162 @@
...
-renders an inflated brain surface repeatedly, rotating it by 5 degrees
-around the *z* axis between redraws. At each
-revolution, the time spent for the full revolution is reported on
-the terminal window where mne_opengl_test was
-started from. The program renders the surface until the interrupt
-key (usually control-c) is pressed on the terminal window.
-
-mne_opengl_test is located
-in the ``bin`` directory and is thus started as:
+If MNE-C can't find ``libgfortran.so.1``, you can probably safely link that
```
Comment (drammock) [inline doc/install_mne_c.rst]: this paragraph+codeblock is all new.  Needs scrutiny.

File: doc/install_mne_python.rst [hunk 14]
```diff
@@ -2,101 +2,112 @@
...
+    scipy:         1.1.0
     matplotlib:    3.0.2 {backend=Qt5Agg}
 
     sklearn:       0.20.2
-    nibabel:       2.3.3
-    mayavi:        4.7.0.dev0 {qt_api=pyqt5, PyQt5=5.10.1}
+    nibabel:       2.3.1
+    mayavi:        4.7.0.dev0 {qt_api=pyqt5, PyQt5=5.9.2}
     cupy:          Not found
-    pandas:        0.24.0
```
Comment (larsoner) [inline doc/install_mne_python.rst]: please revert these changes to the output, I had it (recently) set to what you'd get with the latest MNE env.

File: doc/pre_install.rst [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+and extends considerably the analysis and visualization capabilities.
+MNE-python is collaboratively developed and has more than 150 contributors.
+
+The :ref:`ch_matlab` provides a MATLAB interface to the .fif file format and
+other MNE data structures, and provides example MATLAB implementations of
+some of the core analysis functionality of MNE-C. It is distributed alongside
+MNE-C, and can also be downloaded from the `MNE-MATLAB git repository`_.
+
+:ref:`mne_cpp` provides core MNE functionality implemented in C++ and is
+primarily intended for embedded and real-time applications.
```
Comment (larsoner) [inline doc/pre_install.rst]: I would make these bullet points, each starting with the name of the package (add MNE-C as a bullet)

File: doc/pre_install.rst [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+managing MNE projects so that they comply with the Brain Imaging Data Structure
+specification (`MNE-BIDS`_).
+
+What should I install?
+^^^^^^^^^^^^^^^^^^^^^^
+
+If you intend only to perform ERP, ERF, or other sensor-level analyses,
+:doc:`MNE-python <install_mne_python>` is all you need. If you
+prefer MATLAB over python, probably all you need is :doc:`MNE-C
+<install_mne_c>` — the MNE MATLAB toolbox is distributed with it.
```
Comment (larsoner) [inline doc/pre_install.rst]: Somehow you should note that MNE-C+MNE-MATLAB has fewer features than MNE-Python.

File: doc/pre_install.rst [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+If you want to transform sensor recordings into estimates of localized brain
+activity, you will most likely need:
+
+- :doc:`FreeSurfer <install_freesurfer>` (to convert structural MRI scans into
+  models of the scalp, inner/outer skull, and cortical surfaces)
```
Comment (larsoner) [inline doc/pre_install.rst]: no need for parens on these bullets

File: doc/install_mne_c.rst [hunk 18]
```diff
@@ -2,290 +2,162 @@
...
 .. code-block:: console
 
     $ echo $SHELL
 
-If the output indicates a POSIX shell ``bash`` or ``sh`` then you should export
-variables and source the bash/sh setup file with three commands:
+To configure MNE-C automatically for ``bash`` or ``sh`` shells, add this to
+your ``.bashrc``::
 
-.. code-block:: console
```
[the remark this answers is not in the document]
Comment (drammock) [reply doc/install_mne_c.rst]: reason for doing this was that we're telling people to put these into their `.bashrc` (or `.zshrc`, etc), rather than type into their interactive shell... I thought people might wrongly copy-paste the leading `$`.  But I can put it back if you think it actually helps.

[the remark this answers is not in the document]
Comment (larsoner) [reply doc/install_mne_c.rst]: Then it would be `code-block:: bash`. The default linter is `Python` for `::` (and IIRC it just decides "is Python" or "is not Python", in this case it will do the latter which is not as good as "is Bash", which the `code-block` will do).

But FWIW we used CSS to disable the `$` copying, so they physically can't do it anymore. So I'm fine with `code-block:: bash` or `code-block:: console`.

[the remark this answers is not in the document]
Comment (drammock) [reply doc/install_mne_c.rst]: Really?  I can select-copy-paste the `$` on this page of the current `dev` docs: https://mne-tools.github.io/dev/install_mne_python.html

Either way, `code-block:: bash` is better than what I've currently done.

[the remark this answers is not in the document]
Comment (larsoner) [reply doc/install_mne_c.rst]: Here is my CTRL-A (same thing when using mouse):

screenshot from 2019-02-08 16-08-59

What browser are you using? Maybe it does not respect `user-select` in CSS. Or maybe you need to hard-reload (shift-F5) the page to get it to reload all resources, including CSS files.

**Comment by agramfort** [comment]
just a quick nitpick. Please write Python with a capital P

>

File: doc/install_mne_c.rst (hunk 18 above)
[the remark this answers is not in the document]
Comment (drammock) [reply doc/install_mne_c.rst]: Here is my Ctrl+A after force-reload of both `style.css` (which I opened in its own tab for that purpose) and of the page in question.

mne-select-all

Browser is firefox 65.0 for Linux...  I did an `apt upgrade` yesterday so it's as current as they come.

[the remark this answers is not in the document]
Comment (larsoner) [reply doc/install_mne_c.rst]: Fix in #5911

**Comment by drammock** [comment]
@agramfort do you want capital P also in "MNE-python"?  Or just for "Python" by itself?

**Comment by larsoner** [comment]
Both

**Comment by drammock** [comment]
oops, should have googled `[skip travis]` and `[skip appveyor]` before pushing...

**Comment by drammock** [comment]
latest artifacts: https://11164-1301584-gh.circle-artifacts.com/0/dev/getting_started.html

**Comment by drammock** [comment]
latest artifacts: https://11166-1301584-gh.circle-artifacts.com/0/dev/getting_started.html

@larsoner I think I've addressed all your requests / suggestions, but happy to wait for feedback from @choldgraf or others.

File: doc/install_mne_c.rst [hunk 19]
```diff
@@ -2,290 +2,175 @@
...
+  For installation on macOS, you also need:
 
-Installing from a compressed tar archive
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  - the `XCode developer tools`_.
+  - an X Window System such as XQuartz_. Specifically, you need version 2.7.9
+    of XQuartz; the most current version (2.7.11, as of Feb. 2019) will not
+    work.
+  - the netpbm_ library. The recommended way to get netpbm is to install
+    MacPorts_, and then run ``sudo port install netpbm`` in the Terminal app.
```
Comment (agramfort) [inline doc/install_mne_c.rst]: macports is pretty dead now. Is this still up to date? needed? Would brew do the job too?

File: doc/install_mne_c.rst [hunk 20]
```diff
@@ -2,290 +2,175 @@
...
-subdirecotries in the MNE distribution. Or the command
-"ls $MNE_ROOT/bin/mne_setup_sh" should succeed if you have set MNE_ROOT
-correctly.
+    export MNE_ROOT=<path_to_MNE>
+    export MATLAB_ROOT=<path_to_MATLAB>
+    source $MNE_ROOT/bin/mne_setup_sh
 
-If you do not have a Matlab distribution, then leave MATLAB_ROOT undefined.
+where ``<path_to_MNE>`` and ``<path_to_MATLAB>`` are replaced by the absolute
+paths to MNE-C and MATLAB, respectively.
```
Comment (agramfort) [inline doc/install_mne_c.rst]: I would repeat here that matlab is clearly optional.

File: doc/install_mne_c.rst [hunk 21]
```diff
@@ -2,290 +2,175 @@
...
-    | ``SUBJECTS_DIR``        | Location of the MRI data.                  |
-    +-------------------------+--------------------------------------------+
-    | ``SUBJECT``             | Name of the current subject.               |
-    +-------------------------+--------------------------------------------+
-    | ``MNE_TRIGGER_CH_NAME`` | Name of the trigger channel in raw data,   |
-    |                         | see :ref:`mne_process_raw`.                |
-    +-------------------------+--------------------------------------------+
-    | ``MNE_TRIGGER_CH_MASK`` | Mask to be applied to the trigger channel  |
-    |                         | values, see :ref:`mne_process_raw`.        |
-    +-------------------------+--------------------------------------------+
```
Comment (agramfort) [inline doc/install_mne_c.rst]: MNE_TRIGGER_CH_MASK and MNE_TRIGGER_CH_NAME variables are not documented anymore?

**Comment by agramfort** [comment]
this looks great. If I would make one tiny comment it's that it is not obvious how arrive at Getting Started when you land on documentation page.

what came out of our latest (aborted) doc sprint is that the top content in https://11166-1301584-gh.circle-artifacts.com/0/dev/documentation.html lacks clear visual highlights to say where to start, ie what is important what is not. I would try to limit the number of links, make the getting started link support obvious. What should be there for me is:

Getting started
Frequency asked questions
Getting help section
How to "Cite MNE"
What's new

could this also be part of the PR?

File: doc/install_mne_c.rst (hunk 19 above)
Comment (drammock) [reply doc/install_mne_c.rst]: I didn't know macports was dead... they had a release 4 months ago and their github repo's last change was 29 days ago...  In any case, I don't have any insight into what's best nowadays since I don't have access to a mac computer.

There is [a brew formula for netpbm](https://brewformulas.org/Netpbm), so if someone else can confirm that it works (or if you sufficiently trust that `brew install netpbm` will work) then I can change the instructions, no problem.

Comment (agramfort) [reply doc/install_mne_c.rst]: I confirm that I use

brew install netpbm

and that it works for me. Just add that `brew install netpbm` is an alternative and for which sudo should not be needed. Thanks

File: doc/install_mne_c.rst (hunk 21 above)
Comment (drammock) [reply doc/install_mne_c.rst]: fixed in 897cbbb

File: doc/install_mne_c.rst (hunk 20 above)
Comment (drammock) [reply doc/install_mne_c.rst]: fixed in e10bdae

File: doc/install_mne_c.rst (hunk 19 above)
Comment (drammock) [reply doc/install_mne_c.rst]: fixed in b447cc1

**Comment by drammock** [comment]
@agramfort  This PR only tries to deal with pages that fall under the "Install" link in the website header.  I agree that the Documentation landing page needs work.  My plan was to do that in a subsequent PR --- mostly because I think it will be a lot more work than the Install pages, and may take a couple iterations to get it right and make everyone happy.

That said, maybe on the Install landing page, the "getting started" heading text is misleading?  Would it be better to say "Installation - overview" or something like that?

**Comment by agramfort** [comment]
ok fair enough to postpone this to the next PR.

I don't know exactly how to do it but seeing new users last week made me
realise it's not easy to know where to click first....

>

**Comment by drammock** [comment]
@agramfort @larsoner not sure what's causing the circleci failure (`make html_dev-noplot` succeeds locally)... it looks like a Qt problem, possibly a Qt version conflict (?).  I've pushed another commit to trigger rebuild, but if the problem doesn't fix itself I may need help tracking down what is wrong with the CircleCI recipe.

**Comment by larsoner** [comment]
You need to rebase, it has been fixed in `master` but CircleCI does not build a merged version (anymore -- it used to, but our code stopped doing the merge properly at some point).

**Comment by drammock** [comment]
@larsoner @agramfort rebase and force-pushed worked.  Ready for merge.

**Comment by agramfort** [comment]
thanks a lot @drammock !

**Comment by massich** [comment]
I got late to the party. But this looks great, thx @drammock

**Comment by ktavabi** [comment]
@drammock what version of macOS did you use to test MNE-C and XQuartz?

**Comment by drammock** [comment]
@kambysese I don't have a Mac, that bit of info came from @larsoner.

**Comment by larsoner** [comment]
@ktavabi those instructions worked for me on 10.13, but appear not to on Mojave (10.14).

**Comment by buildqa** [comment]
I will check later on if I can try the the install instructions at,
https://mne-tools.github.io/dev/install_mne_c.html
on a 10.14 system.  I think I have 10.14 installed on an external
drive (which I do not have in front of me right now).

**Comment by ktavabi** [comment]
@drammock I just encountered this issue with `matplotlib:3.0.2 {backend=MacOSX}` 

> ImportError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends. If you are using (Ana)Conda please install python.app and replace the use of 'python' with 'pythonw'. See 'Working with Matplotlib on OSX' in the Matplotlib FAQ for more information.

The fix is documented [here](https://matplotlib.org/faq/osx_framework.html). Maybe worth mentioning in the docs if still relevant.
